src/vs/workbench/contrib/notebook/browser/view/cellParts/cellOutput.ts

```diff
@@ -590,12 +590,14 @@ export class CellOutputContainer extends CellContentPart {
 			clearTimeout(this._outputHeightTimer);
 		}
 
+		const executionState = this._notebookExecutionStateService.getCellExecution(this.viewCell.uri);
+
 		if (synchronous) {
 			this.viewCell.unlockOutputHeight();
-		} else {
+		} else if (executionState?.state !== NotebookCellExecutionState.Executing) {
 			this._outputHeightTimer = setTimeout(() => {
```

Comment:
It's possible that execution state is still `Executing` when we finish the output update (for example, the cell is executing but there is no more outputs). We need a listener on cell execution state change to ensure that we always unlock the output height at the end.